### PR TITLE
Document some distortion world and party functions

### PR DESCRIPTION
**Pull request by ravepossum**
Follow up to PR #207 

This picks off and documents a few functions and constants that are referenced in `field_control.c`, mostly related to the party and the distortion world.

### CODE REVIEW

File: src/player_avatar.c [hunk 1]
```diff
@@ -884,22 +884,26 @@ int PlayerAvatar_DistortionWorldState (PlayerAvatar * const playerAvatar)
...
 {
-    int v0 = PlayerAvatar_DistortionWorldState(playerAvatar);
+    int state = PlayerAvatar_DistortionState(playerAvatar);
 
-    if ((v0 == 0) || (v0 == 1)) {
+    if (state == AVATAR_DISTORTION_STATE_NONE || state == AVATAR_DISTORTION_STATE_ACTIVE) {
         return FALSE;
     }
 
     return TRUE;
```
Comment (lhearachel) [inline src/player_avatar.c:897]: Does this match?

```c
return (state == AVATAR_DISTORTION_STATE_NONE || state == AVATAR_DISTORTION_STATE_ACTIVE) == FALSE;
```

Or, if possible:

```c
return state != AVATAR_DISTORTION_STATE_NONE && state != AVATAR_DISTORTION_STATE_ACTIVE;
```

I'd recommend looking at the ASM here and fiddling with the branch-return sequence to see if you can find something that matches more cleanly.

File: src/player_avatar.c [hunk 2]
```diff
@@ -884,22 +884,26 @@ int PlayerAvatar_DistortionWorldState (PlayerAvatar * const playerAvatar)
...
 }
 
-BOOL sub_0205F16C (PlayerAvatar * const playerAvatar)
+BOOL PlayerAvatar_DistortionStateOnFloor (PlayerAvatar * const playerAvatar)
 {
-    int v0 = PlayerAvatar_DistortionWorldState(playerAvatar);
+    int state = PlayerAvatar_DistortionState(playerAvatar);
 
-    if ((v0 == 0) || (v0 == 1) || (v0 == 2)) {
+    if (state == AVATAR_DISTORTION_STATE_NONE
```
Comment (lhearachel) [inline src/player_avatar.c:904]:
```c
return state == AVATAR_DISTORTION_STATE_NONE
    || state == AVATAR_DISTORTION_STATE_ACTIVE
    || state == AVATAR_DISTORTION_STATE_FLOOR;
```

File: src/unk_02052C6C.c [hunk 3]
```diff
@@ -209,7 +209,7 @@ void sub_02052E58 (TaskManager * param0)
 
     v7 = Party_GetFromSavedata(fieldSystem->saveData);
 
-    sub_02054AC4(v7);
+    Party_SetChampionRibbons(v7);
```
Comment (lhearachel) [inline src/unk_02052C6C.c:212]: [aside] Oh, hey, hall-of-fame code.

File: src/unk_02054884.c [hunk 4]
```diff
@@ -14,17 +14,14 @@
...
 {
-    if (Pokemon_GetValue(param0, MON_DATA_CURRENT_HP, NULL) == 0) {
-        return 0;
-    }
-
-    if (Pokemon_GetValue(param0, MON_DATA_IS_EGG, NULL)) {
-        return 0;
+    // this can be simplified further, but it won't match
+    if (Pokemon_GetValue(mon, MON_DATA_CURRENT_HP, NULL) == 0) {
+        return FALSE;
```
Comment (lhearachel) [inline src/unk_02054884.c:21]: How did you try to simplify this?

File: src/unk_02054884.c [hunk 5]
```diff
@@ -80,79 +77,73 @@ void sub_02054988 (Party * param0, int param1, int param2, u16 param3)
...
+    int i;
+    int partyCount = Party_GetCurrentCount(party);
 
-    for (v0 = 0; v0 < v1; v0++) {
-        v2 = Party_GetPokemonBySlotIndex(param0, v0);
+    for (i = 0; i < partyCount; i++) {
+        Pokemon *mon = Party_GetPokemonBySlotIndex(party, i);
 
-        if (Pokemon_GetValue(v2, MON_DATA_IS_EGG, NULL) != 0) {
+        if (Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL) != 0) {
```
Comment (lhearachel) [inline src/unk_02054884.c:90]:
```suggestion
        if (Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL) != FALSE) {
```

File: src/unk_02054884.c [hunk 6]
```diff
@@ -80,79 +77,73 @@ void sub_02054988 (Party * param0, int param1, int param2, u16 param3)
...
-    v1 = Party_GetCurrentCount(param0);
+    for (i = 0; i < partyCount; i++) {
+        Pokemon *mon = Party_GetPokemonBySlotIndex(party, i);
 
-    for (v0 = 0; v0 < v1; v0++) {
-        v2 = Party_GetPokemonBySlotIndex(param0, v0);
-
-        if (Pokemon_GetValue(v2, MON_DATA_IS_EGG, NULL) == 0) {
-            return v2;
+        if (Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL) == 0) {
```
Comment (lhearachel) [inline src/unk_02054884.c:145]:
```suggestion
        if (Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL) == FALSE) {
```

File: src/unk_02054884.c [hunk 7]
```diff
@@ -80,79 +77,73 @@ void sub_02054988 (Party * param0, int param1, int param2, u16 param3)
...
+        Pokemon *mon = Party_GetPokemonBySlotIndex(party, i);
 
-        if (sub_02054884(v2)) {
-            return v2;
+        if (Pokemon_CanBattle(mon)) {
+            return mon;
         }
     }
 
     GF_ASSERT(0);
```
Comment (lhearachel) [inline src/unk_02054884.c:133]:
```suggestion
    GF_ASSERT(FALSE);
```

File: src/unk_02054884.c [hunk 8]
```diff
@@ -80,79 +77,73 @@ void sub_02054988 (Party * param0, int param1, int param2, u16 param3)
...
+            count++;
         }
     }
 
-    return v2;
+    return count;
 }
 
-Pokemon * Party_FirstBattler (const Party * param0)
+Pokemon * Party_FirstBattler (const Party *party)
```
Comment (lhearachel) [inline src/unk_02054884.c:120]: I feel like the naming between this and `Party_FirstMon` is a little confusing; maybe:

- `Party_FirstBattler` -> `Party_FindFirstEligibleBattler`
- `Party_FirstMon` -> `Party_FindFirstHatchedMon`

File: src/unk_02054884.c [hunk 9]
```diff
@@ -161,22 +152,20 @@ Pokemon * sub_02054A74 (const Party * param0)
...
-void sub_02054AC4 (Party * param0)
+void Party_SetChampionRibbons (Party *party)
 {
-    int v0, v1;
-    u8 v2 = 1;
-    Pokemon * v3;
-
-    v1 = Party_GetCurrentCount(param0);
+    int i;
+    u8 ribbonActive = 1;
```
Comment (lhearachel) [inline src/unk_02054884.c:161]:
```suggestion
    u8 championRibbon = TRUE;
```

File: src/unk_02054884.c [hunk 10]
```diff
@@ -161,22 +152,20 @@ Pokemon * sub_02054A74 (const Party * param0)
 
 BOOL Party_HasTwoAliveMons (const Party *party)
 {
-    return sub_02054A10(party) >= 2;
+    return Party_AliveMonsCount(party) >= 2;
 }
 
-void sub_02054AC4 (Party * param0)
+void Party_SetChampionRibbons (Party *party)
```
Comment (lhearachel) [inline src/unk_02054884.c:158]:
```suggestion
void Party_GiveChampionRibbons (Party *party)
```

My logic here is that `SetChampionRibbons` implies to me that I can specify whether the members have the ribbon or not. `Give` is unambiguous; after this routine, all party members will have the ribbon flag flipped to `TRUE`.

File: src/unk_0205F180.c [hunk 11]
```diff
@@ -861,9 +861,11 @@ static int sub_0205FCC0 (PlayerAvatar * playerAvatar, int param1)
...
-        int v1 = PlayerAvatar_DistortionWorldState(playerAvatar);
+        int distortionState = PlayerAvatar_DistortionState(playerAvatar);
 
-        if ((v1 != 0) && (v1 != 1) && (v1 != 2)) {
+        if (distortionState != AVATAR_DISTORTION_STATE_NONE 
+            && distortionState != AVATAR_DISTORTION_STATE_ACTIVE 
+            && distortionState != AVATAR_DISTORTION_STATE_FLOOR) {
             v0 = PlayerAvatar_GetMoveDir(playerAvatar);
         }
     }
```
Comment (lhearachel) [inline src/unk_0205F180.c:871]: Unnecessary scope block?

**Review by lhearachel — COMMENTED** [review]
As a general comment leading out of discussion on Discord, we're considering moving towards prefixing subroutines with appropriate verbage as a general naming scheme. Feel free to hold off on that until we reach a full consensus on the approach, but I wanted to make sure that I recorded it here.

File: src/unk_02054884.c (hunk 9 above)
Comment (ravepossum) [reply src/unk_02054884.c:161]: The fact that this doesn't match with a `BOOL` instead of `u8` makes me sad.

Comment (lhearachel) [reply src/unk_02054884.c:161]: Memory sizes haha!

We should still make use of the `BOOL` constants where possible, even if the underlying type is not actually `BOOL` (read: `int`).

File: src/unk_02054884.c (hunk 4 above)
Comment (ravepossum) [reply src/unk_02054884.c:21]: I tried these:
```
return !(Pokemon_GetValue(mon, MON_DATA_CURRENT_HP, NULL) == 0) || !Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL);
```

```
return Pokemon_GetValue(mon, MON_DATA_CURRENT_HP, NULL) != 0 || !Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL);
```
```
return !(Pokemon_GetValue(mon, MON_DATA_CURRENT_HP, NULL) == 0 && Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL));
```

File: src/player_avatar.c (hunk 1 above)
Comment (lhearachel) [reply src/player_avatar.c:897]: Did some fiddling and we determined that the feasible match here is:

```c
return PlayerAvatar_DistortionState(playerAvatar) > (u32)AVATAR_DISTORTION_STATE_FLOOR;
```

That's pretty smelly, so we'll keep this sentinel structure instead.

File: src/unk_02054884.c (hunk 4 above)
Comment (lhearachel) [reply src/unk_02054884.c:21]: I took a stab at this and got to this:

```c
BOOL Pokemon_CanBattle(void *mon)
{
    return Pokemon_GetValue(mon, MON_DATA_CURRENT_HP, NULL) != 0
        && !Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL);
}
```

which is logically (and functionally) equivalent, but does not match byte-for-byte. So, meh. Keep this as-is.

File: src/player_avatar.c (hunk 2 above)
Comment (lhearachel) [reply src/player_avatar.c:904]: Does it really not match with this change? Also, why remove this indentation in the most recent revision?

Comment (ravepossum) [reply src/player_avatar.c]: My monkey brain read right past the `return` in your comment and read it as a suggestion to change the indentation there. Not my finest moment.

**Review by lhearachel — APPROVED** [review]
Pre-emptively approving this pending consensus on function name paradigms.

**Comment by lhearachel** [comment]
We reached consensus on the naming scheme for functions moving forward. Please prefix your documented functions in this PR with appropriate verbage, per the naming scheme: `[<Module>_]<Imperative><Object>`

**Review by lhearachel — APPROVED** [review]

**Pull request merged**